package/expo-package/src/optionalDependencies/takePhoto.ts

```diff
@@ -36,44 +43,63 @@ export const takePhoto = ImagePicker
 
         if (permissionGranted) {
           const imagePickerSuccessResult = await ImagePicker.launchCameraAsync({
+            mediaTypes: mediaTypeMap[mediaType],
             quality: Math.min(Math.max(0, compressImageQuality), 1),
           });
           const canceled = imagePickerSuccessResult.canceled;
           const assets = imagePickerSuccessResult.assets;
+          if (canceled || !assets.length) {
+            return { cancelled: true };
+          }
           // since we only support single photo upload for now we will only be focusing on 0'th element.
-          const photo = assets && assets[0];
-
-          if (canceled === false && photo && photo.height && photo.width && photo.uri) {
-            let size: Size = {};
-            if (Platform.OS === 'android') {
-              const getSize = (): Promise<Size> =>
-                new Promise((resolve) => {
-                  Image.getSize(photo.uri, (width, height) => {
-                    resolve({ height, width });
-                  });
-                });
-
-              try {
-                const { height, width } = await getSize();
-                size.height = height;
-                size.width = width;
-              } catch (e) {
-                console.warn('Error get image size of picture caputred from camera ', e);
-              }
-            } else {
-              size = {
-                height: photo.height,
-                width: photo.width,
-              };
-            }
-
+          const photo = assets[0];
+          if (photo.mimeType.includes('video')) {
```

Comment:
This doesn't guarantee that `photo` is not `undefined` or `null`

Reply:
!assets.length is checked on line 51

Reply:
Same deal as [here](https://github.com/GetStream/stream-chat-react-native/pull/2870#discussion_r1916094434)